truss/remote/baseten/auth.py

```diff
@@ -0,0 +1,54 @@
+import functools
+import logging
+import os
+from typing import Optional
+
+from truss.remote.baseten.error import AuthorizationError
+
+logger = logging.getLogger(__name__)
+
+
+class ApiKey:
+    value: str
+
+    def __init__(self, value: str):
+        self.value = value
+
+    def headers(self):
```

Comment:
nit, sounds like this will always be one header, maybe we should call it `header`.